### PR TITLE
chore(release): update readme in prep for deprecation

### PR DESCRIPTION
**Pull request by kmova**
- add release process details
- update build commands
- add migration repo links

Signed-off-by: kmova <kiran.mova@mayadata.io>

### CODE REVIEW

**Comment by kmova** [comment]
This PR needs to be further refactored after merging https://github.com/openebs/maya/pull/1819

(Remove the dependency on openebs/openebs for test script)

File: README.md [hunk 1]
```diff
@@ -1,18 +1,54 @@
...
 
-## Building OpenEBS provisioner from source
+Please refer to our documentation at [OpenEBS Documentation](http://openebs.io/).
 
-Following command will generate `openebs-provisioner` binary. 
+## Release
+
+Prior to creating a release tag on this repository on `v2.12.x` branch with the required fixes, ensure that the dependent data engine repositories and provisioner are tagged. Once the code is merged, use the following sequence to release a new version for the legacy components:
+- (Optional) New release tag on v2.12.x branch of [openebs/linux-utils](https://github.com/openebs/linux-utils)
+- (Optional) New release tag on v0.6.x branch of [openebs/ndm](https://github.com/openebs/ndm)
```
Comment (niladrih) [inline README.md]:
```suggestion
- (Optional) New release tag on v0.6.x branch of [openebs/ndm](https://github.com/openebs/node-disk-manager)
```

**Review by niladrih — COMMENTED** [review]
Broken link

**Review by niladrih — APPROVED** [review]
LGTM

**Pull request merged**